### PR TITLE
automated test step execution optimization

### PR DESCRIPTION
**Pull request by kakj-go**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
Automated test step running uses the last execution record of the current scene to render the ${{ outputs.xxx.xxx }} placeholders in it

##### run scene pre step output
image

##### run step use pre step value

image

##### run step rendered request params value 
image

### CODE REVIEW

File: modules/dop/services/autotest_v2/scene.go [hunk 1]
```diff
@@ -525,6 +525,32 @@ func (svc *Service) ExecuteDiceAutotestSceneStep(req apistructs.AutotestExecuteS
 		apiTestStr = strings.ReplaceAll(apiTestStr, expression.OldLeftPlaceholder+expression.Params+"."+param.Name+expression.OldRightPlaceholder, expression.ReplaceRandomParams(param.Temp))
 	}
 
+	var pipelinePageListRequest = apistructs.PipelinePageListRequest{
+		PageNum:  1,
+		PageSize: 10,
```
Comment (sfwn) [inline modules/dop/services/autotest_v2/scene.go:530]: why get 10 but only use one (index 0)?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dop/services/autotest_v2/scene.go [hunk 2]
```diff
@@ -525,6 +525,32 @@ func (svc *Service) ExecuteDiceAutotestSceneStep(req apistructs.AutotestExecuteS
...
+		if err == nil && pipeline != nil {
+			for _, stage := range pipeline.PipelineStages {
+				for _, task := range stage.PipelineTasks {
+					for _, result := range task.Result.Metadata {
+						apiTestStr = strings.ReplaceAll(apiTestStr, expression.GenOutputRef(task.Name, result.Name), result.Value)
+					}
+				}
+			}
+		}
+	}
```
Comment (Effet) [inline modules/dop/services/autotest_v2/scene.go:552]: too deep, split to a func or multiple funcs

File: modules/dop/services/autotest_v2/scene.go (hunk 1 above)
Comment (kakj-go) [reply modules/dop/services/autotest_v2/scene.go]: Really shouldn't query 10

Get the subscript 0, just to get the latest end record，I added sorting by id should be no problem

**Review by Effet — DISMISSED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**